### PR TITLE
fix(core): Prevent XSS in user update endpoints

### PR DESCRIPTION
**Pull request by ivov**
https://linear.app/n8n/issue/SEC-77/217-insufficient-input-validation

### CODE REVIEW

File: packages/cli/src/GenericHelpers.ts [hunk 1]
```diff
@@ -31,3 +37,36 @@ export async function validateEntity(
...
+class StringWithNoXss {
+	@NoXss()
+	value: string;
+
+	constructor(value: string) {
+		this.value = value;
+	}
+}
+
+export async function validateRecordNoXss(record: Record<string, string>) {
```
Comment (ivov) [inline packages/cli/src/GenericHelpers.ts:50]: Assuming good enough until we have more streamlined payload validation in place. Else let me know.

File: packages/cli/src/controllers/me.controller.ts [hunk 2]
```diff
@@ -237,6 +239,9 @@ export class MeController {
 		const payload = plainToInstance(UserSettingsUpdatePayload, req.body, {
 			excludeExtraneousValues: true,
 		});
+
+		await validateEntity(payload);
```
Comment (tomi) [inline packages/cli/src/controllers/me.controller.ts:243]: 👍

File: packages/cli/src/GenericHelpers.ts (hunk 1 above)
Comment (tomi) [reply packages/cli/src/GenericHelpers.ts:50]: Not a big fan of this approach, but you are right that we first need a standard mechanism for payload validation. Maybe we could add a comment that this is only a temporary solution

**Review by tomi — COMMENTED** [review]
Looks good, couple comments

**Review by tomi — APPROVED** [review]
👏

**Comment by janober** [comment]
Got released with [`n8n@1.55.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.55.0)

**Pull request merged**